File: backend-go/internal/modules/system/controller/online_user_controller.go
package controller

import (
	"drip-admin/backend-go/internal/common"
	"github.com/gin-gonic/gin"
)

var _ = common.ApiResponse{}

// OnlineUsers returns online sessions.
// @Summary List online users
// @Tags online-user
// @Produce json
// @Security ApiKeyAuth
// @Param page query int false "Page"
// @Param pageSize query int false "Page size"
// @Param username query string false "Username"
// @Param ip query string false "IP"
// @Param deviceType query string false "Device type"
// @Success 200 {object} common.ApiResponse
// @Failure 401 {object} common.ApiResponse
// @Failure 403 {object} common.ApiResponse
// @Router /system/onlineUser [get]
func (ctl *Controller) OnlineUsers(c *gin.Context) {
	ctl.svc.OnlineUsers(c)
}

// OnlineUser returns an online session detail.
// @Summary Online user detail
// @Tags online-user
// @Produce json
// @Security ApiKeyAuth
// @Param tokenId path string true "Token ID"
// @Success 200 {object} common.ApiResponse
// @Failure 401 {object} common.ApiResponse
// @Failure 403 {object} common.ApiResponse
// @Failure 404 {object} common.ApiResponse
// @Router /system/onlineUser/{tokenId} [get]
func (ctl *Controller) OnlineUser(c *gin.Context) {
	ctl.svc.OnlineUser(c)
}

// KickoutOnlineUser kicks out an online session.
// @Summary Kick out online user
// @Tags online-user
// @Produce json
// @Security ApiKeyAuth
// @Param tokenId path string true "Token ID"
// @Success 200 {object} common.ApiResponse
// @Failure 401 {object} common.ApiResponse
// @Failure 403 {object} common.ApiResponse
// @Failure 404 {object} common.ApiResponse
// @Router /system/onlineUser/{tokenId}/kickout [post]
func (ctl *Controller) KickoutOnlineUser(c *gin.Context) {
	ctl.svc.KickoutOnlineUser(c)
}

File: backend-python/app/modules/system/controller/online_user_controller.py
from fastapi import APIRouter, Depends, Query, Request

from app.common.api_response import ApiResponse, success
from app.common.pagination import PageQuery
from app.modules.system.router import require_permission
from app.modules.system.service.online_user_service import OnlineUserService

router = APIRouter(tags=["online-user"])


def service(request: Request) -> OnlineUserService:
    return OnlineUserService(request.app.state.redis)


@router.get("/onlineUser", response_model=ApiResponse)
async def online_users(
    request: Request,
    page: PageQuery = Depends(),
    username: str | None = None,
    ip: str | None = None,
    device_type: str | None = Query(default=None, alias="deviceType"),
    _: dict = Depends(require_permission("system:online:list")),
    current: OnlineUserService = Depends(service),
) -> ApiResponse:
    token = request.headers.get(request.app.state.settings.token.name, "")
    return success(await current.list_users(page, token, username, ip, device_type))


@router.get("/onlineUser/{tokenId}", response_model=ApiResponse)
async def online_user_detail(
    tokenId: str,
    request: Request,
    _: dict = Depends(require_permission("system:online:list")),
    current: OnlineUserService = Depends(service),
) -> ApiResponse:
    token = request.headers.get(request.app.state.settings.token.name, "")
    return success(await current.detail(tokenId, token))


@router.post("/onlineUser/{tokenId}/kickout", response_model=ApiResponse)
async def kickout_online_user(
    tokenId: str,
    request: Request,
    _: dict = Depends(require_permission("system:online:kickout")),
    current: OnlineUserService = Depends(service),
) -> ApiResponse:
    token = request.headers.get(request.app.state.settings.token.name, "")
    await current.kickout(tokenId, token)
    return success()
